lib/notifications/notification_server.ex

```diff
@@ -0,0 +1,116 @@
+defmodule Notifications.NotificationServer do
+  use GenServer
+
+  alias Notifications.Notification
+  alias Realtime.BlockWaiver
+  alias Schedule.Block
+
+  require Logger
+
+  @type __MODULE__ :: {schedule_data_mod :: module()}
+
+  defstruct [:schedule_data_mod]
+
+  # Client
+
+  @spec default_name() :: GenServer.name()
+  def default_name(), do: Notifications.NotificationServer
+
+  @spec start_link(Keyword.t()) :: GenServer.on_start()
+  def start_link(opts \\ []) do
+    name = Keyword.get(opts, :name, __MODULE__)
+    schedule_data_mod = Keyword.get(opts, :schedule_data_mod, Schedule)
+    GenServer.start_link(__MODULE__, %{schedule_data_mod: schedule_data_mod}, name: name)
+  end
+
+  @spec new_block_waivers(BlockWaiver.block_waivers_by_block_key(), GenServer.server()) :: :ok
+  def new_block_waivers(new_waivers_by_block_key, server \\ default_name()) do
+    GenServer.cast(server, {:new_block_waivers, new_waivers_by_block_key})
+  end
+
+  # Server
+
+  @impl GenServer
+  def init(initial_state) do
+    schedule_data_mod = Map.fetch!(initial_state, :schedule_data_mod)
+    {:ok, %__MODULE__{schedule_data_mod: schedule_data_mod}}
+  end
+
+  @impl true
+  def handle_cast({:new_block_waivers, new_block_waivers_by_block_key}, state) do
+    new_notifications =
+      new_block_waivers_by_block_key
+      |> convert_new_block_waivers_to_notifications(state.schedule_data_mod)
+
+    if !Enum.empty?(new_notifications) do
+      Logger.warn(
```

Comment:
Notifications are getting logged in dev, however, it doesn't seem to be filling in certain of the fields correctly:

<img width="873" alt="Screen Shot 2020-08-14 at 6 09 42 AM" src="https://user-images.githubusercontent.com/9827/90239029-e8759480-ddf4-11ea-905a-23bff3ba6df5.png">

Reply:
Confusion between timestamps and times-of-day, it looks like.